torch_xla/core/dynamo_bridge.py

```diff
@@ -126,18 +128,36 @@ def _args_on_cuda(input_args: tuple) -> bool:
 
   return input_device.type == "cuda"
 
+def _move_xla_cuda_tensor_to_cuda(tensor):
+  assert tensor.device.type == "xla", "The tensor is not an XLA tensor"
+  is_xla_cuda = True if xu.getenv_as("PJRT_DEVICE", str, "").lower() == "cuda" else False
+  assert is_xla_cuda, "The XLA tensor is not on CUDA"
+  # consumer is torch, producer is torch_xla
+
+  # Similar logic as torch.utils.dlpack.from_dlpack
+  # https://github.com/pytorch/pytorch/blob/b0ef363972203b163cddc95e4c6054b8221c2300/torch/utils/dlpack.py#L114-L115
+  device_id = tensor.device.index
+  stream = torch_xla._XLAC._get_stream_for_cuda_device(device_id)
+  stream = 1 if stream == 0 else stream
+  assert stream is None or type(stream) is int
+  external_stream = torch.cuda.ExternalStream(stream)
+  current_stream = torch.cuda.current_stream()
+  if external_stream != current_stream:
+    event = torch.cuda.Event()
+    event.record(current_stream)
+    external_stream.wait_event(event)
+  dlpack = torch_xla_dlpack.to_dlpack(tensor)
+  cuda_tensor = torch.utils.dlpack.from_dlpack(dlpack)
+  return cuda_tensor
```

Comment:
I feel like this would be useful outside the dynamo bridge, too. What do you think about moving this to `torch_xla.dlpack.as_eager()` or something like this?

Reply:
Done.
OTOH, the logic here borrows from https://github.com/pytorch/pytorch/blob/b0ef363972203b163cddc95e4c6054b8221c2300/torch/utils/dlpack.py#L114-L115. So if https://github.com/pytorch/pytorch/pull/128176 gets merged, we don't need this logic here anymore.